### PR TITLE
task for 3 module - create new pull request

### PR DESCRIPTION
**Pull request by OlenaEpam**

### CODE REVIEW

File: hw_3/task_for_homework_3.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE.
+
+
+
+last iz TO calculate nuMber OF Whitespace characteRS in this Tex. caREFULL, not only Spaces, 
+but ALL whitespaces. I got 87.
+"""
+# all text letters make lower-case
+text_lower = text_sample.lower()
+# split text by words
```
Comment (BulbaYim) [inline hw_3/task_for_homework_3.py:22]: Better stay 1 empty line code before a new comment, it more readable 

```
# all text letters make lower-case
text_lower = text_sample.lower()

# split text by words
c = text_lower.split()

# check every word is it is equal to 'iz' and replace it to 'is'
for i in c:
    if i == 'iz':
        index_of_iz = c.index(i)
        c[index_of_iz] = 'is'
```

File: hw_3/task_for_homework_3.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+c = text_lower.split()
+# check every word is it is equal to 'iz' and replace it to 'is'
+for i in c:
```
Comment (BulbaYim) [inline hw_3/task_for_homework_3.py]: E.x `for word in word_list`

the same for the following loops

It's bad practice to use one character symbols for variable names, but use for index, math symbols is okay

File: hw_3/task_for_homework_3.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        last_words.append(last_word)
+# join all words received and capitalize the sentence
+new_sentence = ' '.join(last_words).capitalize()
+# every sentence strip of spaces and capitalize the sentence (if not to strip, we can by index
+# take 2nd element and capitalize this element only)
+for i in string_newest:
+    b = i.strip().capitalize()
+    # replace sentences from small letter to sentences from big letter
+    index_of_sentence = string_newest.index(i)
+    string_newest[index_of_sentence] = b
```
Comment (BulbaYim) [inline hw_3/task_for_homework_3.py]: there is possible to use list comprehension 
`string_newest = [sentence.strip().capitalize() for sentence in string_newest]`

File: hw_3/task_for_homework_3.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+string_new = ' '.join(c)
+# split this new string into sentences
+string_newest = string_new.split('.')
+last_words = []
+# further split sentences into list of words and retrive last word from element in the list
+for i in string_newest:
+    if i:
+        words = i.split()
+        last_word = words[-1]
+        last_words.append(last_word)
```
Comment (BulbaYim) [inline hw_3/task_for_homework_3.py]: `[row.split()[-1] for row in string_newest if row]`

**Review by BulbaYim — APPROVED** [review]

File: hw_3/task_for_homework_3.py (hunk 1 above)
Comment (OlenaEpam) [reply hw_3/task_for_homework_3.py:22]: Thanks. Changed that in next commit https://github.com/OlenaEpam/data-repo/pull/7/commits/de9a2857cf7f4eb95e929b00962d37de3d3786d5

File: hw_3/task_for_homework_3.py (hunk 2 above)
Comment (OlenaEpam) [reply hw_3/task_for_homework_3.py]: Changed that in next commit https://github.com/OlenaEpam/data-repo/pull/7/commits/de9a2857cf7f4eb95e929b00962d37de3d3786d5

File: hw_3/task_for_homework_3.py (hunk 3 above)
Comment (OlenaEpam) [reply hw_3/task_for_homework_3.py]: Understood, thanks https://github.com/OlenaEpam/data-repo/pull/7/commits/de9a2857cf7f4eb95e929b00962d37de3d3786d5

File: hw_3/task_for_homework_3.py (hunk 4 above)
Comment (OlenaEpam) [reply hw_3/task_for_homework_3.py]: Rewrote here https://github.com/OlenaEpam/data-repo/pull/7/commits/de9a2857cf7f4eb95e929b00962d37de3d3786d5

**Pull request merged**